cmds/internet/main.go

```diff
@@ -124,10 +126,16 @@ func configureZOS() error {
 			return errors.Wrapf(err, "could not bring %s up", zosChild)
 		}
 
-		log.Info().Msg("writing udhcp init service")
+		log.Info().Msg("writing dhcpcd init service")
+
+		// terminate older udhcpc service if found
+		if err := z.Terminate(dhcpService, "udhcpc"); err != nil {
+			log.Error().Err(err).Msg("fail to terminate older dhcp-zos (udhcpc) zinit service")
+			return err
+		}
```

Comment:
Terminate is not enough. since zinit still know the service was that specific name, replacing the zinit yaml file will not override the service configuration in zinit memory.
in other words, when u terminate a service, then change the file then start it will still start with the original config (before replacing the file)
Instead, what you need to do is to `terminate`, `forget`, `change the config file`, then `monitor`.

That being said, the init service does not live forever, it starts one time on system boot and is included in the basic zos image. It means that internet service does NOT need to check or replace the config, because there is nothing before it starts, which means it can directly use the dhcpcd service. 

It's up to `networkd` which can be updated during the machine runtime to check if the dhcp service is in fact using the dhcpcd binary, otherwise it then should terminate, forget, repalce, and monitor using the new config.

Reply:
I just noticed you added the new `Terminate` function to which does exactly that. which is nice, but i would rename it to something like `Destroy` instead of terminate.

I still think `internet` does not need to care since it's one time thing that starts on boot, hence there is no service before already.